xbmc/pvr/PVRDatabase.cpp

```diff
@@ -488,36 +490,17 @@ int CPVRDatabase::Get(CPVRChannelGroup &group)
 
 bool CPVRDatabase::PersistChannels(CPVRChannelGroup &group)
 {
-  bool bReturn(true);
+  for (auto &member : group.m_members) {
+    const CPVRChannelPtr &channel = member.second.channel;
 
-  for (PVR_CHANNEL_GROUP_MEMBERS::iterator it = group.m_members.begin(); it != group.m_members.end(); ++it)
-  {
-    if (it->second.channel->IsChanged() || it->second.channel->IsNew())
```

Comment:
channel is a reference to a smart pointer? looks strange. what is the goal of doing this?